### PR TITLE
docs(databricks): add guide for Databricks integration

### PR DESCRIPTION
**Pull request by rexledesma**
### Summary & Motivation
Here, we provide the basic template for using Dagster and Databricks, using the resource abstraction.

Note that we do not yet provide a template for using a prebuilt asset factory to generate these assets. This will be added later. For now, add value by showing how the integration is set up in the first place. 

### How I Tested These Changes
local, eyes

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12150** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12150" target="_blank">Graphite</a> 
    * **PR #12153** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12153" target="_blank">Graphite</a> 
      * **PR #12326** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12326" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12326?utm_source=stack-comment).

**Review by sryza — COMMENTED** [review]
I think that we should message the `databricks_pyspark_step_launcher` as an equally-valid way of running Dagster with Databricks.

I haven't done a scientific study, but I suspect that a slight majority of our databricks users are using the step launcher.

If I were personally using Dagster with Databricks, I would try to use the step launcher (unless I was using Scala).

File: docs/content/integrations/databricks.mdx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+databricks_client_instance = databricks_client.configured(
+    {
+        "host": {"env": "DATABRICKS_HOST"},
+        "token": {"env": "DATABRICKS_TOKEN"},
+    }
+)
+```
+
+## Step 2: Create an asset from a Databricks job
```
Comment (erinkcochran87) [inline docs/content/integrations/databricks.mdx:47]:
```suggestion
---

## Step 2: Create an asset from a Databricks job
```

File: docs/content/integrations/databricks.mdx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        python_params=None,
+        spark_submit_params=None,
+    )
+
+@asset(non_argument_deps={"my_databricks_table"})
+def my_downstream_asset() -> None:
+    ...
+```
+
+## Step 3: Schedule a Databricks job
```
Comment (erinkcochran87) [inline docs/content/integrations/databricks.mdx:79]:
```suggestion
---

## Step 3: Schedule a Databricks job
```

File: docs/content/integrations/databricks.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        ScheduleDefinition(
+            job=materialize_databricks_table_and_downstream,
+            cron_schedule="@daily",
+        )
+    ],
+    resources={"databricks": databricks_client_instance},
+)
+```
+
+## What's next?
```
Comment (erinkcochran87) [inline docs/content/integrations/databricks.mdx:108]:
```suggestion
---

## What's next?
```

File: docs/content/integrations/databricks.mdx [hunk 4]
```diff
@@ -0,0 +1,115 @@
+---
+title: "Using Dagster with Databricks"
```
Comment (erinkcochran87) [inline docs/content/integrations/databricks.mdx:2]:
```suggestion
title: "Using Dagster with Databricks | Dagster Docs"
```

File: docs/content/integrations/databricks.mdx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+2. A `token` corresponding to a personal access token for your Databricks workspace, stored in an environment variable `DATABRICKS_TOKEN`.
+3. A `DATABRICKS_JOB_ID` for the Databricks job you want to run.
+
+You can follow the [Databricks API authentication instructions](https://docs.databricks.com/dev-tools/python-api.html#step-1-set-up-authentication) to retrieve these values.
+
+---
+
+## Step 1: Connecting to Databricks
+
+The first step in using Databricks with Dagster is to tell Dagster how to connect to your Databricks workspace using a Databricks [resource](/concepts/resources). This resource contains information on where your Databricks workspace is located and any credentials sourced from environment variables that are needed to access it.
```
Comment (erinkcochran87) [inline docs/content/integrations/databricks.mdx:34]: Should we add a link to the `dagster-databricks` API docs in here? Maybe to `/_apidocs/libraries/dagster-databricks#dagster_databricks.DatabricksClient`?

File: docs/content/integrations/databricks.mdx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    {
+        "host": {"env": "DATABRICKS_HOST"},
+        "token": {"env": "DATABRICKS_TOKEN"},
+    }
+)
+```
+
+## Step 2: Create an asset from a Databricks job
+
+In this step, we model a Databricks job as a software-defined asset that launches a run of the Databricks job to materialize a Databricks table.
```
Comment (erinkcochran87) [inline docs/content/integrations/databricks.mdx:49]:
```suggestion
In this step, we model a Databricks job as a [software-defined asset](/concepts/assets/software-defined-assets) that launches a run of the Databricks job to materialize a Databricks table.
```

File: docs/content/integrations/databricks.mdx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    )
+
+@asset(non_argument_deps={"my_databricks_table"})
+def my_downstream_asset() -> None:
+    ...
+```
+
+## Step 3: Schedule a Databricks job
+
+Now that your Databricks job is modeled as a Dagster asset, you can schedule it to run on a regular cadence. In this example, we schedule the asset and its downstream dependencies to materialize every day. When the asset materializes, the underlying Databricks job will be triggered to refresh the asset.
```
Comment (erinkcochran87) [inline docs/content/integrations/databricks.mdx:81]: Add a link to relevant Dagster schedule documentation?

**Review by erinkcochran87 — APPROVED** [review]
A handful of comments on small stuff, but overall this lgtm

File: docs/content/integrations/databricks.mdx [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    create_databricks_run_now_op,
+    create_databricks_submit_run_op,
+)
+
+my_databricks_run_now_op = create_databricks_run_now_op(
+    databricks_job_id=DATABRICKS_JOB_ID
+)
+
+my_databricks_submit_run_op = create_databricks_submit_run_op(
+    databricks_job_configuration={
```
[the remark this answers is not in the document]
Comment (rexledesma) [reply docs/content/integrations/databricks.mdx:77]: This is covered in the linked `dagster-databricks` API docs. It's basically the keyword arguments to the the underlying Databricks API call, in dictionary form.

File: docs/content/integrations/databricks.mdx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+```python startafter=start_schedule_databricks endbefore=end_schedule_databricks file=/integrations/databricks/databricks.py dedent=4
+from dagster import (
+    AssetSelection,
+    Definitions,
+    ScheduleDefinition,
+    define_asset_job,
+)
+
+materialize_databricks_table = define_asset_job(
+    name="materialize_databricks_table",
```
[the remark this answers is not in the document]
Comment (rexledesma) [reply docs/content/integrations/databricks.mdx]: I've moved this to step 2, since this is part of constructing the job from the op/asset anyways.

**Pull request merged**